backend/pkg/api/data_access/vdb_rocket_pool.go

```diff
@@ -2,14 +2,251 @@ package dataaccess
 
 import (
 	"context"
+	"encoding/hex"
+	"fmt"
+	"slices"
+	"strings"
 
+	"github.com/doug-martin/goqu/v9"
 	"github.com/gobitfly/beaconchain/pkg/api/enums"
 	t "github.com/gobitfly/beaconchain/pkg/api/types"
+	"github.com/gobitfly/beaconchain/pkg/commons/utils"
+	"github.com/shopspring/decimal"
+	"golang.org/x/sync/errgroup"
 )
 
 func (d *DataAccessService) GetValidatorDashboardRocketPool(ctx context.Context, dashboardId t.VDBId, cursor string, colSort t.Sort[enums.VDBRocketPoolColumn], search string, limit uint64) ([]t.VDBRocketPoolTableRow, *t.Paging, error) {
-	// TODO @DATA-ACCESS
-	return d.dummy.GetValidatorDashboardRocketPool(ctx, dashboardId, cursor, colSort, search, limit)
+	// Initialize the cursor
+	var currentCursor t.RocketPoolCursor
+	var err error
+	var paging t.Paging
+
+	if cursor != "" {
+		currentCursor, err = utils.StringToCursor[t.RocketPoolCursor](cursor)
+		if err != nil {
+			return nil, &paging, fmt.Errorf("failed to parse passed cursor as ValidatorsCursor: %w", err)
+		}
+	}
+
+	type RocketPoolData struct {
+		Node                   []byte          `db:"naddress"`
+		StakedETH              decimal.Decimal `db:"staked_eth"`
+		StakedRPL              decimal.Decimal `db:"rpl_stake"`
+		MinipoolsTotal         uint64          `db:"minipools_total"`
+		MinipoolsLeb16         uint64          `db:"minipools_leb16"`
+		MinipoolsLeb8          uint64          `db:"minipools_leb8"`
+		AvgCommission          float64         `db:"avg_commission"`
+		RPLClaimed             decimal.Decimal `db:"rpl_claimed"`
+		RPLUnclaimed           decimal.Decimal `db:"rpl_unclaimed"`
+		EffectiveRPL           decimal.Decimal `db:"effective_rpl"`
+		SmoothingPoolOptIn     bool            `db:"smoothing_pool_opted_in"`
+		SmoothingPoolClaimed   decimal.Decimal `db:"claimed_smoothing_pool"`
+		SmoothingPoolUnclaimed decimal.Decimal `db:"unclaimed_smoothing_pool"`
+		Timezone               string          `db:"timezone_location"`
+		RefundBalance          decimal.Decimal `db:"refund_balance"`
+		DepositCredit          decimal.Decimal `db:"deposit_credit"`
+		RPLStakeMin            decimal.Decimal `db:"min_rpl_stake"`
+		RPLStakeMax            decimal.Decimal `db:"max_rpl_stake"`
+	}
+
+	rocketPoolResults := []RocketPoolData{}
+
+	ds := goqu.Dialect("postgres").
+		From(goqu.T("rocketpool_minipools").As("mp")).
+		Select(
+			goqu.T("n").Col("address").As("naddress"),
+			goqu.L("SUM(mp.node_deposit_balance) AS staked_eth"),
+			goqu.T("n").Col("rpl_stake"),
+			goqu.L("COUNT(mp.address) as minipools_total"),
+			goqu.L("SUM(CASE WHEN mp.node_deposit_balance = 8e18 THEN 1 ELSE 0 END) AS minipools_leb8"),
+			goqu.L("SUM(CASE WHEN mp.node_deposit_balance = 16e18 THEN 1 ELSE 0 END) AS minipools_leb16"),
+			goqu.L("CASE WHEN COUNT(mp.address) > 0 THEN SUM(mp.node_fee) / COUNT(mp.address) ELSE 0 END AS avg_commission"),
```

Comment:
**check:** Should the average commission only factor in `Staking` minipools? Depending on how the data is used. Also this could be weighted by the amount of borrowed ETH per MP, as the user is prob interested on the avg additional rewards.

Reply:
Great catch, only counting staking minipools now and weighted based on deposit size

Reply:
> weighted based on deposit size

Not sure what it's doing now.. am I reading this right, basically it's `fee * node_deposit_balance / node_deposit_balance`? That'd not achieve a lot, but I could be missing something :smile:
I was thinking about `fee * user_deposit_balance / node_deposit_balance`. But I'm pretty sure there are other places where this has not been applied, should prob also check that and create tickets maybe.